### ISSUE TITLE
How could I submit a job in Galaxy with pbs runner

### ISSUE THREAD

**Issue by Brentbin** [galaxyproject/galaxy#2580]
It's my job_config.xml

```
<?xml version="1.0"?>
<!-- A sample job config that explicitly configures job running the way it is configured by default (if there is no explicit config). -->
<job_conf>
    <plugins>
       <plugin id="pbs" type="runner" load="galaxy.jobs.runners.pbs:PBSJobRunner" workers="4"/>
    </plugins>
    <handlers default="handlers">
          <handler id="pbs_1" tags="handlers"/>
     </handlers>
    <destinations default="pbs_def">
        <destination id="pbs_def" runner="pbs"/>
        <!--<destination id="long_jobs" runner="pbs" tage="long_job"/>
           <param id="Resource_List">walltime=72:00:00,nodes=1:ppn=8</param>
        </destination>-->
    </destinations>
</job_conf>
```

Thx

**Comment by Brentbin** [comment]
And the info of debug
`Traceback (most recent call last):
  File "/mnt/ceph/galaxy/lib/galaxy/webapps/galaxy/buildapp.py", line 63, in paste_app_factory
    app = galaxy.app.UniverseApplication( global_conf=global_conf, **kwargs )
  File "/mnt/ceph/galaxy/lib/galaxy/app.py", line 174, in __init__
    self.job_manager = manager.JobManager( self )
  File "/mnt/ceph/galaxy/lib/galaxy/jobs/manager.py", line 23, in __init__
    self.job_handler = handler.JobHandler( app )
  File "/mnt/ceph/galaxy/lib/galaxy/jobs/handler.py", line 32, in __init__
    self.dispatcher = DefaultJobDispatcher( app )
  File "/mnt/ceph/galaxy/lib/galaxy/jobs/handler.py", line 723, in __init__
    self.job_runners = self.app.job_config.get_job_runner_plugins( self.app.config.server_name )
  File "/mnt/ceph/galaxy/lib/galaxy/jobs/__init__.py", line 680, in get_job_runner_plugins
    rval[id] = runner_class( self.app, runner[ 'workers' ], **runner.get( 'kwds', {} ) )
  File "/mnt/ceph/galaxy/lib/galaxy/jobs/runners/pbs.py", line 93, in __init__
    assert pbs is not None, PBS_IMPORT_MESSAGE
AssertionError: The Python pbs-python package is required to use this feature, please install it or correct the following error:
ImportError /git/galaxy/.venv/lib/python2.7/site-packages/pbs/_pbs.so: undefined symbol: pbs_submit_hash_ext`

**Comment by Brentbin** [comment]
But the stat of job is always
`This job is waiting to run`

**Comment by Brentbin** [comment]
I found that 
NO RUNNER of job
in job manager

**Comment by pvanheus** [comment]
Hi @Brentbin 

Firstly, this question is perhaps best directed at the galaxy-dev mailing list. However, since it is here.

1) Do you have a handler called pbs defined in your galaxy.ini ? I.e. is there a [server:pbs] section. If not, you might want to change that to direct to your existing server (typically id="main" in a new Galaxy install).

2) Is the PBS runner starting correctly? It loads the Python pbs modules - this should be available in the virtualenv that Galaxy uses (by default GALAXY_HOME/.venv, where GALAXY_HOME is the location where Galaxy is installed). If it doesn't load correct there will be a message in your log starting with "The Python pbs-python package is required to use this feature".

**Comment by Brentbin** [comment]
Thx for your attention @pvanheus 
I have install and run a pbs in my pbs_1 server
this is my galaxy.ini server config block

```
[server:pbs_1]
use = egg:Paster#http
host=192.168.0.190
port=15003
use_threadpool = True
threadpool_kill_thread_limit = 10800
```

and this is my job_config.xml

```
<job_conf>
    <plugins>
       <plugin id="pbs_1" type="runner" load="galaxy.jobs.runners.pbs:PBSJobRunner" workers="4">
        </plugin>
    </plugins>
    <handlers default="handlers">
          <handler id="pbs_1" tags="handlers"/>
     </handlers>
    <destinations default="pbs_def">
        <destination id="pbs_def" runner="pbs"/>
    </destinations>
</job_conf>
```

the stat of pbs is 

```
master
     state = free
     power_state = Running
     np = 1
     ntype = cluster
     status = rectime=1467796962,macaddr=00:10:18:1a:f0:a3,cpuclock=Fixed,varattr=,jobs=,state=free,netload=7582574040,gres=,loadave=0.00,ncpus=4,physmem=16267932kb,availmem=23344160kb,totmem=24525464kb,idletime=24104,nusers=0,nsessions=0,uname=Linux master 3.10.0-327.22.2.el7.x86_64 #1 SMP Thu Jun 23 17:05:11 UTC 2016 x86_64,opsys=linux
     mom_service_port = 15002
     mom_manager_port = 15003
```

the debug info is

```
galaxy.tools.actions.upload DEBUG 2016-07-06 03:50:03,823 Checked and cleaned uploads (4990.305 ms)
galaxy.tools.actions.upload_common INFO 2016-07-06 03:50:05,011 tool upload1 created job id 49
galaxy.tools.actions.upload DEBUG 2016-07-06 03:50:06,032 Created upload job (2207.478 ms)
galaxy.tools.execute DEBUG 2016-07-06 03:50:06,032 Tool [upload1] created job [49] (7243.392 ms)
galaxy.tools.execute DEBUG 2016-07-06 03:50:06,074 Executed 1 job(s) for tool upload1 request: (7326.827 ms)
```

However the stat of job is still "This job is waiting to run"

**Comment by hexylena** [comment]
Hi @Brentbin 

``` xml
<job_conf>
    <plugins>
       <plugin id="pbs_1" type="runner" load="galaxy.jobs.runners.pbs:PBSJobRunner" workers="4">
        </plugin>
    </plugins>
    <handlers default="handlers">
          <handler id="pbs_1" tags="handlers"/>
     </handlers>
    <destinations default="pbs_def">
        <destination id="pbs_def" runner="pbs"/>
    </destinations>
</job_conf>
```

your destinations configure where your jobs default to going. You've defined them as going by default to `pbs_def`. However, pbs_def doesn't exist as a plugin destination. Try changing the definitions block to:

``` xml
<destinations default="pbs_1">
</destinations>
```
